source/main/dqm4hep-mysql-push-parameters.cc

```diff
@@ -0,0 +1,201 @@
+ * Parses the different directories and populate the
+ * qtest file with found objects.
+ */
+
+// -- dqm4hep headers
+#include <dqm4hep/StatusCodes.h>
+#include <dqm4hep/Internal.h>
+#include <dqm4hep/DBInterface.h>
+#include <dqm4hep/XMLParser.h>
+
+// -- tclap headers
+#include "tclap/CmdLine.h"
+#include "tclap/Arg.h"
+
+using namespace std;
+using namespace dqm4hep::core;
+
+struct DBInterfaceMetadata
+{
+  std::shared_ptr<DBInterface> m_dbInterface;
+  std::string      m_host;
+  std::string      m_user;
+  std::string      m_db;
+  std::string      m_password;
+};
+
+typedef std::map<std::string, DBInterfaceMetadata> DBInterfaceMetadataMap; 
+
+int main(int argc, char* argv[])
+{
+  std::string cmdLineFooter = "Please report bug to <dqm4hep@gmail.com>";
+  std::unique_ptr<TCLAP::CmdLine> pCommandLine(new TCLAP::CmdLine(cmdLineFooter, ' ', DQMCore_VERSION_STR));
+
+  TCLAP::ValueArg<std::string> xmlInputFileArg(
+      "f"
+      , "parameter-file"
+      , "The xml input file describing the parameters and databases to feed"
+      , true
+      , ""
+      , "string");
+  pCommandLine->add(xmlInputFileArg);
+
+  // parse command line
+  pCommandLine->parse(argc, argv);
+  
+  XMLParser parser;
+  parser.parse(xmlInputFileArg.getValue());
+  
+  TiXmlDocument &document(parser.document());
+  DBInterfaceMetadataMap dbInterfaceMap;
+
+  TiXmlElement *root = document.RootElement();
+  TiXmlElement *databases = root->FirstChildElement("databases");
+    
+  if(!databases)
+  {
+    dqm_error( "ERROR: no <databases> tag found in root element !" );
+    return 1;
+  }
+  
+  for(TiXmlElement *db = databases->FirstChildElement("db") ; db ; db = db->NextSiblingElement("db"))
+  {
+    const char *dbID = db->Attribute("id");
+    
+    if(!dbID || std::string(dbID).empty())
+    {
+      dqm_error( "ERROR: <db> tag has no 'id' attribute or is empty, line {0}", db->Row() );
+      return 1;
+    }
+    
+    if(dbInterfaceMap.find(dbID) != dbInterfaceMap.end())
+    {
+      dqm_error( "ERROR: <db> tag with id '{0}', line {1}, is duplicated !", dbID, db->Row() );
+      return 1;
+    }
+    
+    std::string dbHost = db->Attribute("host") ? db->Attribute("host") : "localhost";
+    std::string dbUser = db->Attribute("user") ? db->Attribute("user") : "DQM4HEP";
+    std::string dbDb = db->Attribute("db") ? db->Attribute("db") : "DQM4HEP";    
+    std::string dbPassword = db->Attribute("password") ? db->Attribute("password") : "";
+
+    DBInterfaceMetadata metadata;
+    metadata.m_dbInterface = std::make_shared<DBInterface>();
+    metadata.m_host = dbHost;
+    metadata.m_user = dbUser;
+    metadata.m_db = dbDb;
+    metadata.m_password = dbPassword;
+    
+    dbInterfaceMap[dbID] = metadata; 
+  }
+  
+  int nParameters(0);
+  std::set<std::pair<std::string,std::string>> dbTables;
+  
+  for(TiXmlElement *parameters = root->FirstChildElement("parameters") ; parameters ; parameters = parameters->NextSiblingElement("parameters"))
```

Comment:
Same as before. but this time if no `<parameters>` tag are found user won't get any message except that it `successfully pushed 0 parameter(s) into 0 db table(s)`